### PR TITLE
Added dynamic typed

### PR DESCRIPTION
**Pull request by agamkrbit**
Made some changes to implement dynamic typed for variation values
- Made the `EppoValueDeserializer` deserialiser a bit lenient so that it doesn't throw error in case of unexpected values. It will return `null` in that case. Also, added a case for `OBJECT` and `POJO` typed values.
- Updated the `Variation` DTO to use `typedValue` instead of `value` and `ExperimentConfiguration` DTO to use `typedOverrides` instead of `overrides`
- Added some new get assignment function to fetch the typed values.
  - getStringAssignment
  - getBooleanAssignment
  - getLongAssignment
  - getJSONAssignment (which will return the json string)
- Updated the test cases to pull the latest sample experiment configuration

### CODE REVIEW

File: src/main/java/com/eppo/sdk/EppoClient.java [hunk 1]
```diff
@@ -121,16 +122,136 @@ public Optional<String> getAssignment(
...
+        Optional<EppoValue> value = this.getAssignment(subjectKey, experimentKey, subjectAttributes);
+        if (value.isEmpty()) {
+            return Optional.empty();
+        }
+
+        switch (type) {
+            case BOOLEAN:
+                return Optional.of(value.get().boolValue());
+            case NUMBER:
+                return Optional.of(value.get().longValue());
```
Comment (aarsilv) [inline src/main/java/com/eppo/sdk/EppoClient.java:147]: Do we not have to worry about numbers with decimals? Looking at the existing code, it seems we don't worry about them, but I figure I'd flag it.

File: src/main/java/com/eppo/sdk/EppoClient.java [hunk 2]
```diff
@@ -121,16 +122,136 @@ public Optional<String> getAssignment(
...
+    }
+
+    /**
+     * This function will return boolean assignment value
+     * @param subjectKey
+     * @param experimentKey
+     * @param subjectAttributes
+     * @return
+     */
+    public Optional<Boolean> getBooleanAssignment(String subjectKey, String experimentKey, SubjectAttributes subjectAttributes) {
```
Comment (aarsilv) [inline src/main/java/com/eppo/sdk/EppoClient.java:192]: I think the non-string `get<Type>Assignments()` functions don't have unit tests. While the code is simple/sound, I'd consider whipping up a few more test cases to protect future us.

**Review by aarsilv — APPROVED** [review]
Yay types! Approving as-is with some minor comments you can take or leave as you see fit!

File: src/main/java/com/eppo/sdk/EppoClient.java (hunk 1 above)
Comment (agamkrbit) [reply src/main/java/com/eppo/sdk/EppoClient.java:147]: @aarsilv Thanks for flagging this. I will change it to decimal.

**Review by leoromanovsky — APPROVED** [review]

**Pull request merged**